src/cache/mst_cache.rs

```diff
@@ -0,0 +1,21 @@
+use std::fmt;
+use crate::model::model;
+
+pub struct MST_Cache {
```

Comment:
I think according to [Rust naming conventions](https://doc.rust-lang.org/1.0.0/style/style/naming/README.html) it should be `MstCache`. Not a major point, but still I think we should follow them.